core/status/viable.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package status
+
+// IsMachineViable returns true if the machine is started.
+func IsMachineViable(status StatusInfo) bool {
+	// This traps the known machine status codes, but if the status isn't
+	// recognised, we assume the machine is not viable.
+	switch status.Status {
+	case Started:
+		return true
+	case Pending, Down, Stopped, Error, Unknown:
+		return false
+	default:
+		return false
+	}
+}
+
+// IsInstanceViable returns true if the instance is running.
+func IsInstanceViable(status StatusInfo) bool {
+	// This traps the known instance status codes, but if the status isn't
+	// recognised, we assume the instance is not viable.
+	switch status.Status {
+	case Running:
+		return true
+	case Empty, Allocating, Error, ProvisioningError, Unknown:
+		return false
+	default:
+		return false
+	}
+}
+
+// IsAgentViable returns true if the agent is idle or executing.
+func IsAgentViable(status StatusInfo) bool {
+	// This traps the known agent status codes, but if the status isn't
+	// recognised, we assume the agent is not viable.
+	switch status.Status {
+	case Idle, Executing:
+		return true
+	case Allocating, Error, Failed, Rebooting:
+		return false
+	default:
+		return false
+	}
+}
+
+// IsUnitWorkloadViable returns true if the unit workload is active, or is
+// in a state where it is expected to become active.
+func IsUnitWorkloadViable(status StatusInfo) bool {
```

Comment:
Can we add a TODO to rework this. It should be all about the agent. The workload should be irrelevant. But up to install is done we're using workload status to surface the state of things (mainly for the user).

Reply:
Yeah, I think when we do model migration, we should think long and hard about this piece.

Reply:
Checking the actual status message (the message intended for human consumption) is especially evil here. This really needs to go is possible